### ISSUE TITLE
status doesn't work for kube-addons and for kube-master-addons daemon

### ISSUE THREAD

**Issue by marekbiskup** [GoogleCloudPlatform/kubernetes#8803]
```kube-addons``` and ```kube-master-addons``` are created as (bash) daemons. 

The status command for neither of them works. For ```kube-addons```, this is how we check if the process is running:
```
KUBE_ADDONS_SH=/etc/kubernetes/kube-addons.sh
SCRIPTNAME=/etc/init.d/$NAME
...
        ps --no-headers ${pid} | grep ${SCRIPTNAME} > /dev/null || exit $?
```

while the process that is started is ```$KUBE_ADDONS_SH```, not ```$SCRIPTNAME```:
```
root@kubernetes-master:/home/biskup# ps --no-headers 21476 | grep kube-addons
21476 pts/1    S      0:00 /bin/bash /etc/kubernetes/kube-addons.sh
```
Similarly for kube-master-addons.

```kube-addons``` does this when it starts:
```
# We stay around so that status checks by salt make it look like
# the service is good. (We could do this is other ways, but this
# is simple.)
sleep infinity
```
while ```kube-master-addons``` does not.

I suggest also to make ```kube-addons``` and ```kube-master-addons``` consistent, i.e. both should ```sleep infinity``` or both should not. Or, if a different behavior is needed, please comment in the code.

As bash scripts can get easily broken, I suggest also to write a test for the status command (just invoke ```service kube-addons status``` and ```service kube-master-addons status``` at start of some e2e test).

**Comment by marekbiskup** [comment]
Being fixed by #9128

**Comment by marekbiskup** [comment]
Merged in #9128
